### PR TITLE
Feature/backend sse

### PR DESCRIPTION
**Pull request by Cmatrix1**
A base to use sse inside the project
Not connected to Fastapi section

### CODE REVIEW

File: fastcrawler_ui/core/fastapi/app.py [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 def get_application(router=get_routers(), base_app=FastAPI) -> FastAPI:
     app = base_app()
+    origins = ["*"]
+    app.add_middleware(
+        CORSMiddleware,
+        allow_origins=origins,
+        allow_credentials=True,
+        allow_methods=["*"],
+        allow_headers=["*"],
+    )
```
Comment (ManiMozaffar) [inline fastcrawler_ui/core/fastapi/app.py]: Not really a good approach!

File: fastcrawler_ui/routers/sse.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+from fastapi import APIRouter
+
+from uuid import uuid4
+import datetime
+import asyncio
+import random
+import json
+
+
+async def generate_random_log():
```
Comment (ManiMozaffar) [inline fastcrawler_ui/routers/sse.py:11]: if this is intended to be used as "mock" then create another package, and put it there. 
also speficy this with like #TODO: remove mock later

**Review by ManiMozaffar — CHANGES_REQUESTED** [review]
There is few issues with this current implementation;

https://python.plainenglish.io/server-sent-events-for-push-notifications-on-fastapi-73e7ac4a2c2e
check implementation as suggested in here;

1. Uvicorn is not gracefully shut down -> https://stackoverflow.com/questions/58133694/graceful-shutdown-of-uvicorn-starlette-app-with-websockets
2. Responses must not be compressed by ANY middleware in order to work

that's why you should use this lib:
https://github.com/sysid/sse-starlette

P.S: you don't need to use redis, just create the listen_to_channel as send generator.

File: fastcrawler_ui/core/fastapi/app.py (hunk 1 above)
Comment (amirbahador-hub) [reply fastcrawler_ui/core/fastapi/app.py]: in what sense? and what's the alternative?

File: fastcrawler_ui/routers/sse.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                'started_at': started_at,
+                'duration': '12:23',
+                'state': 'Active',
+                }]
+            }
+        yield json.dumps(data)
+        await asyncio.sleep(1)
+
+router = APIRouter()
+
```
Comment (amirbahador-hub) [inline fastcrawler_ui/routers/sse.py:103]: use separation of concerns
we should be able to easily remove mock functions

**Review by amirbahador-hub — CHANGES_REQUESTED** [review]

**Pull request merged**